pyls/workspace.py

```diff
@@ -112,8 +112,8 @@ def _create_document(self, doc_uri, source=None, version=None):
 
 class Document(object):
 
-    def __init__(self, uri, source=None, version=None, local=True, extra_sys_path=None, rope_project_builder=None,
-                 config=None, workspace=None):
+    def __init__(self, uri, workspace, source=None, version=None, local=True, extra_sys_path=None,
+                 rope_project_builder=None, config=None):
```

Comment:
Is it guaranteed that `Document()`  is not instantiated directly by downstream projects or did you just change the API, so to speak?

Reply:
I think it's not possible to have a document without a workspace and that that was the case before only to simplify the tests.

However, I took a quick look to the most important third-party plugins of this project (pyls-my and pyls-black) and they don't seem to use `Document` directly.